doc.js

```diff
@@ -220,6 +235,78 @@ define('doc', ['event'], function(event) {
 				return query(result);
 			},
 
+			'closest' : function(selector) {
+				// only works with non-composite selectors. e.g. '.class' or '#id' or 'div', but not: '#id .class' or 'ul.class'
+				var elements = [];
+				var selectorType;
+				var replacedSelector = selector;
+
+				if (matcher.isTag(selector)) {
+					selectorType = 'tag';
+				} else if (matcher.isClass(selector)) {
+					replacedSelector = selector.replace('.', '');
+					selectorType = 'class';
+				} else if (matcher.isId(selector)) {
+					replacedSelector = selector.replace('#', '');
+					selectorType = 'id';
+				} else {
+					console.warn('You are using composite selectors. e.g. "#id .class" or "tag.class". This selector will only work on Chrome 41+, Firefox 35+ or Opera 28+');
+				}
+
+				var checkForClosestParent = function(el) {
+					var element = el.first();
```

Comment:
@mottam 

`el` is never null because this inner function argument is always a doc object (even if it has length 0).